static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx

```diff
@@ -168,7 +176,7 @@ export function LineChartListWidget(props: Props) {
       },
       transform: transformEventsRequestToArea,
     };
-  }, [props.eventView.query, field, props.organization.slug, selectedListIndex]);
+  }, [props.eventView, field, props.organization.slug, selectedListIndex]);
```

Comment:
Can't say with confidence that I have a strong grasp on what this will do but won't this trigger a re-render on every location change because `eventView` is re-derived each time, resulting in a new reference?

Reply:
Yeah it schedules a rerender, but eventView is no longer being passed to the query component, so the component itself never refires. Re-renders are fine, but re-firing the api request is not. The eventView has more than just the query to look for (in this case statsPeriod was changing), I'd rather not figure out primitives for every facet of eventView that could change and just let the existing component refetch mechanism deal with it.